lib/fetching/content-services/facebook-content-service.js

```diff
@@ -26,116 +27,139 @@ FacebookContentService.prototype._doFetch = function(options, callback) {
-      self.emit('error', new Error('Invalid Facebook URL'));
-      callback([]);
-    } else {
-      var posts = _.findWhere(res.data, {name: 'posts'}).fql_result_set;
-      var comments = _.findWhere(res.data, {name: 'comments'}).fql_result_set;
-      var authors = _.findWhere(res.data, {name: 'authors'}).fql_result_set;
-      callback(self._handleResults(posts, comments, authors));
-    }
-  });
+  if (this._lastReportDate) {
+    params.since = this._lastReportDate.getTime() / 1000;
+  }
+  // Grab posts off of sources feed
+  self._doGet('/v2.2/' + source + '/feed', params, data, callback);
+
+  //If no new posts, grab comments off of most recent post
+  if((!data || data.length == 0) && this.posts[source] != null) {
+    self._doGet('/v2.2/' + this.posts[source].id + '/comments?order=reverse_chronological', params, data, callback);
+  }
 };
 
-// Runs the FQL request
-// queries - The queries to run
-// callback(err, data) - The callback via which to return the resulting data.
-FacebookContentService.prototype._doRequest = function(queries, callback) {
-  graph.fql(queries, function(err, res) {
-    if (err)
-      callback(err, []);
-    else
-      callback(null, res);
+FacebookContentService.prototype._doGet = function(url, params, data, callback) {
+  var self = this;
+
+  graph.get(url , params, function(err, res) {
+    if (res.data && res.data.length > 0) {
+      data = data.concat(res.data);
+    }
+    if (res.paging && res.paging.next && params.since) {
+      self._doGet(res.paging.next, params, data, callback);
+    } else {
+      callback(err, data);
+    }
   });
 };
 
-// Handle data returned by the FQL query. Returns an array of hashes of report data
-// (does not need to be sorted, parent class handles that).
-FacebookContentService.prototype._handleResults = function(posts, comments, authors) {
+FacebookContentService.prototype._handleResults = function(data, source) {
   var self = this;
+  var oldPost = this.posts[source];
   var reportData = [];
-  var postsById = _.indexBy(posts, 'post_id');
 
-  // Handle each post.
-  posts.forEach(function(post) {
-    reportData.push(self._parse(post, {type: 'post', authors: authors}));
-  });
+  if (!data) {
+    return reportData;
+  }
 
-  // Handle each comment.
-  comments.forEach(function(comment) {
-    reportData.push(self._parse(comment, {type: 'comment', authors: authors, post: postsById[comment.post_id]}));
+  data.forEach(function(object) {
+    //If object has a type then it is a post, else it is a comment
+    if (object.type) {
+      // Handle each post and any comment already attached to it
+      reportData.push(self._parse(object, {type: 'post'}));
+      if (object.comments && object.comments.data && object.comments.data.length > 0 ) {
+        object.comments.data.forEach(function(comment) {
+          reportData.push(self._parse(comment, {type: 'comment', url: object.link}));
+        });
+      }
+    } else {
+      if (oldPost == undefined){
```

Comment:
Some more spacing issues in this file. Please add space after the if condition, and above, please remove space after 0 on line 106.